### DISCUSSION TITLE
Some questions regarding the long execution times of job runs. Can anyone help me with this?

### DISCUSSION THREAD

**Discussion by sqkkyzx** [dagster-io/dagster#24679] [Q&A]
I've just started working with Dagster and I'm noticing that the task execution times are quite long. I'm not sure if this is expected, or if I'm using it incorrectly.

I followed the tutorial [Building a pipeline using asset definitions](https://docs.dagster.io/tutorial) for my first attempt, but I added an additional asset to test things out:

```python
@asset
def test() -> None:
    print('A test.')
```

I then manually materialized it, and according to the Events, the entire process from `RUN_START` to `RUN_SUCCESS` took 22.3 seconds, with `STEP_WORKER_STARTING` to `STEP_WORKER_STARTED` alone taking over 10 seconds, and the `STEP_START` to `STEP_SUCCESS` related to the test taking 2.5 seconds.

I'm having a hard time understanding why it would take so long to execute a simple print function. Have I made a mistake somewhere?

![image](https://github.com/user-attachments/assets/15e771cc-0113-47b4-a820-994d7c35d4f6)

**Comment by garethbrickman** [comment]
That does seem to take far too long. I see you're using "hackernews_job" to perform this so I wonder if there's other dependencies being setup to fulfill that job rather than just materializing that single asset without using a job. 

In my testing if I materialize the same asset, or use `define_asset_job` to create an asset job to materialize it, there are no step events and it succeeds virtually instantaneously.

**Reply by sqkkyzx** [reply]
I found that after deploying to Docker, this issue no longer occurs. It might be related to WSL
